### PR TITLE
Restructured messagebroker

### PR DESCRIPTION
**Pull request by janetchen1**
Plugins now interact with global messagebroker living in messagebroker.js, as opposed to app.js inheriting MessageBroker class.

### CODE REVIEW

**Comment by janetchen1** [comment]
Done

File: datastore.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        return current[keys[keys.length - 1]];
+    }
+}
+
+DataStore.prototype.remove = function (keys) {
+    var current = this.dataStore;
+
+    if (!lo.has(this.dataStore, keys)) {
```
Comment (waisbrot) [inline datastore.js]: Did you mean `_` instead of `lo` here?

Comment (janetchen1) [inline datastore.js]: yes, thanks. Corrected that.

File: datastore.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        console.log('BAD KEY TO DS.REMOVE');
+    }
+    else {
+        for (var i = 0; i < keys.length - 1; i++) {
+
+            if (!current[keys[i]]) {
+                console.log('BAD KEYS TO DS.GET');
+            }
```
Comment (waisbrot) [inline datastore.js]: This seems redundant with the above check. When would `has(this.dataStore, keys)` be true, but `current[keys[i]]` be false?

Also, you're not stopping the loop, so you'll continue to try to descend into the object even though it doesn't exist. Is this just a debugging log statement?

File: messagebroker.js [hunk 3]
```diff
@@ -24,8 +24,32 @@ MessageBroker.prototype.init = function () {
     setInterval(this.sendMessage.bind(this), 1500);
 };
 
+var messageBroker;
+
+function initialize(sock) {
+    messageBroker = new MessageBroker(sock);
```
Comment (waisbrot) [inline messagebroker.js]: You might want to throw an exception here if `messageBroker` is already defined, so that you can't accidentally overwrite it.

**Pull request merged**